server/client_server.go

```diff
@@ -22,10 +30,117 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
-// GetCreateMeta returns the metadata needed to implement the UI and validation of
+type ProjectIssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values"`
+}
+
+type FieldInfo struct {
+	Values []map[string]interface{} `json:"values"`
+}
+
+type JiraServerVersionInfo struct {
+	Version string `json:"version"`
+}
+
+// GetIssueInfo returns the issues information based on project id.
+func (client jiraServerClient) GetIssueInfo(projectID string) (*ProjectIssueInfo, *jira.Response, error) {
+	apiEndpoint := fmt.Sprintf("%s%s/issuetypes", APIEndpointCreateIssueMeta, projectID)
+	req, err := client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	issues := ProjectIssueInfo{}
+	response, err := client.Jira.Do(req, &issues)
+	return &issues, response, err
+}
+
+func (client jiraServerClient) GetProjectInfoForPivotJiraVersion(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, *jira.Response, error) {
+	var issueInfo *ProjectIssueInfo
+	var req *http.Request
+
+	projectList, resp, err := client.Jira.Project.ListWithOptions(options)
+	meta := new(jira.CreateMetaInfo)
+
+	if err == nil {
+		for _, proj := range *projectList {
+			meta.Expand = proj.Expand
+			issueInfo, resp, err = client.GetIssueInfo(proj.ID)
+			if err != nil {
+				break
+			}
+
+			for _, issueType := range issueInfo.Values {
+				apiEndpoint := fmt.Sprintf("%s%s/issuetypes/%s", APIEndpointCreateIssueMeta, proj.ID, issueType.Id)
+				req, err = client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
+				if err != nil {
+					break
+				}
+
+				fieldInfo := FieldInfo{}
+				resp, err = client.Jira.Do(req, &fieldInfo)
+				if err != nil {
+					break
+				}
+
+				fieldMap := make(map[string]interface{})
+				for _, fieldValue := range fieldInfo.Values {
+					fieldMap[fmt.Sprintf("%v", fieldValue["fieldId"])] = fieldValue
+				}
+				issueType.Fields = fieldMap
+			}
+			project := &jira.MetaProject{
+				Expand:     proj.Expand,
+				Self:       proj.Self,
+				Id:         proj.ID,
+				Key:        proj.Key,
+				Name:       proj.Name,
+				IssueTypes: issueInfo.Values,
+			}
+
+			meta.Projects = append(meta.Projects, project)
+		}
+	}
+	return meta, resp, err
+}
+
+func (client jiraServerClient) GetProjectInfo(currentVersion, pivotVersion semver.Version, options *jira.GetQueryOptions) (*jira.CreateMetaInfo, *jira.Response, error) {
+	var info *jira.CreateMetaInfo
+	var resp *jira.Response
+	var err error
+
+	if currentVersion.LT(pivotVersion) {
+		info, resp, err = client.Jira.Issue.GetCreateMetaWithOptions(options)
+	} else {
+		info, resp, err = client.GetProjectInfoForPivotJiraVersion(options)
+	}
+	return info, resp, err
```

Comment:
This function can be a bit shorter

```suggestion
	if currentVersion.LT(pivotVersion) {
		return client.Jira.Issue.GetCreateMetaWithOptions(options)
	}

	return client.GetProjectInfoForPivotJiraVersion(options)
```